tests/aws_utils.py

```diff
@@ -276,6 +277,16 @@ class MockValidationResultFactory(Mock):
 # Utility functions
 
 
+def wait_for_s3_key(bucket_name: str, key: str, s3_client: S3Client) -> None:
+
+    process_timeout = datetime.now() + timedelta(minutes=3)
+    while (
+        "Contents" not in s3_client.list_objects(Bucket=bucket_name, Prefix=key)
+        and datetime.now() < process_timeout
+    ):
+        time.sleep(5)
```

Comment:
Should there be an `assert datetime.now() < process_timeout` instead, so that the test fails "early" (before trying to interact with the S3 file) if there is no such key?